buildSrc/src/main/groovy/io/spring/gradle/RemoteProjectTask.groovy

```diff
@@ -0,0 +1,76 @@
+package io.spring.gradle
+
+import groovy.transform.CompileStatic
+import groovy.transform.TypeChecked
+import groovy.transform.TypeCheckingMode
+import org.gradle.api.DefaultTask
+import org.gradle.api.Task
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.OutputDirectory
+import org.gradle.api.tasks.TaskAction
+
+/**
+ * Checkout a project template from a git repository.
+ */
+@CompileStatic
+abstract class RemoteProjectTask extends DefaultTask {
+
+    /**
+     * URI of the git repository.
+     *
+     * Either the remote git repository URL, the path to a local bare git repository or the path to a local git clone.
+     */
+    @Input
+    abstract Property<String> getRemoteUri()
```

Comment:
It might be a nice addition for this to automatically prefix the URL with `https://github.com`, if it's not already present. Then configuration could do `spring-projects/spring-security-samples` instead.